File: Python/Backtracking/combination.py
def combine(n,k):
    # write code here
    res = []
    def helper(start,curr):
        # base
        if len(curr) == k:
            res.append(curr[:])
            return
        for j in range(start,n+1):
            curr.append(j)
            helper(j+1,curr)
            curr.pop()

    helper(1,[])
    return res

File: TypeScript/Backtracking/combination.ts
function combine(n: number, k: number): number[][] {
    const res: number[][] = [];

    function helper(start: number, curr: number[]) {
        if (curr.length === k) {
            res.push([...curr]); // clone array
            return;
        }

        for (let j = start; j <= n; j++) {
            curr.push(j);
            helper(j + 1, curr);
            curr.pop();
        }
    }

    helper(1, []);
    return res;
}